pkg/homedir/homedir_unix.go

```diff
@@ -115,33 +124,30 @@ func GetConfigHome() (string, error) {
 				rootlessConfigHomeDirError = fmt.Errorf("cannot resolve %s: %w", home, err)
 				return
 			}
-			tmpDir := filepath.Join(resolvedHome, ".config")
-			_ = os.MkdirAll(tmpDir, 0o700)
-			st, err := os.Stat(tmpDir)
-			if err != nil {
-				rootlessConfigHomeDirError = err
-				return
-			} else if int(st.Sys().(*syscall.Stat_t).Uid) == os.Geteuid() {
-				cfgHomeDir = tmpDir
-			} else {
-				rootlessConfigHomeDirError = fmt.Errorf("path %q exists and it is not owned by the current user", tmpDir)
-				return
-			}
+			cfgHomeDir = filepath.Join(resolvedHome, ".config")
+		}
+		st, err := os.Stat(cfgHomeDir)
```

Comment:
I am still not convinced that the stat adds value here.

In fact if we keep the the stat then I see zero reason to change anything here because then most callers still have to ignore ENOENT errors which adds a lot of code and doesn't help the issue https://github.com/containers/podman/issues/23818.

Neither HomeDir() or GetCacheHome() validate permissions either.